core/src/Revolution/Sources/modS3MediaSource.php

```diff
@@ -292,16 +294,206 @@ public function getObjectUrl($object = '')
      */
     protected function buildManagerImagePreview($path, $ext, $width, $height, $bases, $properties = [])
     {
-        if ($image = $this->getObjectUrl($path)) {
-            if ($this->getVisibility($path) !== Visibility::PUBLIC) {
-                $image = false;
-            }
-        }
-
         return [
-            'src' => $image,
+            'src' => $this->getObjectUrl($path),
             'width' => $width,
             'height' => $height,
         ];
     }
+
+    /**
+     * Return an array of files and folders at this current level in the directory structure
+     *
+     * @param string $path
+     *
+     * @return array
+     */
+    public function getContainerList($path)
+    {
+        $properties = $this->getPropertyListWithDefaults();
+        $path = $this->postfixSlash($path);
+        if ($path == DIRECTORY_SEPARATOR || $path == '\\') {
+            $path = '';
+        }
+
+        $bases = $this->getBases($path);
+        $imageExtensions = explode(',', $properties['imageExtensions']);
+        $skipFiles = $this->getSkipFilesArray($properties);
+        $allowedExtensions = $this->getAllowedExtensionsArray($properties);
+
+        $directories = $dirNames = $files = $fileNames = [];
+
+        try {
```

Comment:
This is a really big try/catch block. It would be nice if it wasn't this big.

Reply:
This is from the base modMediaSource->getContainerList() function. The try was for the flysystem->ListContents(), so I've wrapped it more closely.